lightkurve/periodogram.py

```diff
@@ -530,8 +533,8 @@ def __repr__(self):
     def from_lightcurve(lc, min_frequency=None, max_frequency=None,
                         min_period=None, max_period=None,
                         frequency=None, period=None,
-                        nterms=1, nyquist_factor=1, oversample_factor=1,
-                        freq_unit=1/u.day, **kwargs):
+                        nterms=1, nyquist_factor=1, oversample_factor=2,
```

Comment:
Question: did you intentionally change the default `oversample_factor` value from `1` to `2`?  Can you explain in a few words why this is a better default?

Reply:
Hi guys, I did indeed change the `oversample_factor` on purpose. Although I can not speak for exoplanets, in asteroseismology, an oversample factor of 1 corresponds to critical sampling, which I believe is usually only done with solar-like oscillators.

In cases where the mode lifetime is much shorter than the duration observations (i.e. they can be resolved into Lorentzians) it doesn't do any harm to use a critically sampled spectrum. This was a pretty common scenario with _Kepler_ data. But now with _TESS_ we've seen that even some of the solar-like oscillators are not being resolved fully, such as nu Indi in the 1 month data. 

The image I attached in my original comment is a 𝛿 Scuti with coherent modes lasting longer than the 4 yr time-base of observations (common for 𝛿 Scts). Because the spectrum is critically sampled, many of the peaks are not sampled at their exact maximum and their amplitudes (and frequencies) are slightly off. For cases like these, to obtain accurate readings of the frequency and amplitude you'd need to do either some kind of interpolation, or simply oversample the spectrum instead.

The only loss incurred by oversampling is that the frequency bins are no longer statistically independent, and the obvious extra computing budget required. I've spoken to Tim Bedding regarding this and he has also suggested oversampling the spectrum by default. He has actually suggested that it should go up to 5 as a trade-off between proper sampling of frequencies & computing time.

To answer your comment below, @christinahedges,
> Please can you tell me how you determined what the true amplitudes are for your target? If this is just to match another code, I'll need a concrete reason before I change the default.

The true (or at least very close to true) amplitudes were obtained by correct sampling of the frequency grid (oversampling :) ). In the below plot I've zoomed in on the central frequency from my original comment. You can see that with `oversample_factor=1` the frequency grid is insufficient to correctly pick out the right frequency (and consequently, the amplitude is underestimated). In fact, even  `oversample_factor=2` does not fully capture the mode.
![oversample_example 1](https://user-images.githubusercontent.com/38233719/52263284-d6ef3900-2982-11e9-87ef-aa5f58bb8199.png)

It is straightforward to show that this is the case by using some artificial data (with known amplitudes) and running it through different oversample factors

Sorry for the very long response to this one point, please let me know if you have any more questions! 

Finally, there's a nice (if not mathematically dense) discussion of oversampling in [this paper](https://academic.oup.com/mnras/article/388/4/1693/981666).

Reply:
Hi, I just noticed this discussion and thought I might chime in on the issue of oversampling. Within asteroseismology there are definitely going to be two different schools of thought here. Those interested in solar-like oscillators typically fit features in the frequency domain, where oversampling really complicates the distribution of power and associated statistics. For classical oscillators (also for binaries), where signals are often assumed to be coherent throughout the time series, oversampling (usually by a factor of 5-10) and amplitude normalization allows one to quickly read off precise signal properties directly from the periodogram (which are probably then refined by fits to the time series).  This decision should be explicitly chosen by the user based on their specific science case.  I think a good default is oversampling=1 since this requires the least computation time, it yields an easy-to-interpret Chi^2 2 degrees-of-freedom noise distribution in the power spectrum, and it matches the default normalization of 'psd'. 
I caution that the heights of oversampled peaks in amplitude spectra risk misinterpretation when the assumption that the underlying signal is coherent in phase, frequency, and amplitude throughout the time series is not valid, making it a risky default.

Since amplitude normalization and oversampling often go hand-in-hand, perhaps it would be a good idea to issue a warning when a periodogram is computed with normalization='amplitude' and oversample_factor=1?

Would this be a useful thing to add a tutorial about?  I'd be happy to work on that when I find some spare time...

Thanks @danielhey for adding this normalization option.  I've had it on my own to-do list for a while, and now it's one less thing... :smile:

Reply:
I agree with Keaton on all counts, especially

>Since amplitude normalization and oversampling often go hand-in-hand, perhaps it would be a good idea to issue a warning when a periodogram is computed with normalization='amplitude' and oversample_factor=1?

I also personally feel like lightkurve should be the path of least resistance for people learning to work with these data or having a quick look at them. The less we complicate things the better.

Thanks for adding your voice to the discussion @keatonb, this back and forth is proving very insightful!

Reply:
Hi all, I hope you don't mind me butting in :-) 
I agree completely that lightkurve should be aimed primarily at those with less experience.  With this in mind, I would suggest a default that gives (i) a properly sampled spectrum, and (ii) units that are easy to interpret.  

(i) for my money, "properly sampled" means oversample_factor>~5.  Note that over- vs critically- is unfortunate choice of names that we are stuck with.  The sampling referred to as "critical" gives a ragged ugly curve (see Dan's plots) that not only makes it hard to extract the true frequencies and amplitudes, it actually throws away some information (happy to explain further).  It is true that some experts like to use critical for solar-like modes that are well-resolved.  But that is a specialist application (and less relevant to TESS than Kepler because of the shorter segments).  Most applications (rotation, pulsations, transits) are much better served by providing a well-sampled spectrum that is a closer approximation to the true spectrum.

(ii) units are a subtle issue (see Appendix A1 of https://ui.adsabs.harvard.edu//#abs/1995A&A...293...87K/abstract ).  Power density is used for solar-like oscillators, especially when measuring noise backgrounds from granulation.  Again, this is a fairly specialised and minority area.  I would think the vast majority of users just want to know the amplitude of the sinusoidal variations.  

There are good reasons why Period04 adopted defaults that give fully-sampled spectra with easy-to-interpret amplitudes, and I strongly encourage lightkurve to do the same.

Reply:
Hi Tim, thanks for adding to the discussion! This is definitely healthy for lightkurve (and educational for me), given that periodogram() was written by somebody who has only studied solar-like oscillators 😬. 

I agree, and I also think a tutorial based around the different use cases discussed in this thread would be an excellent learning tool (although for a future PR, probably)